src/components/cuda/cupti_profiler.c

```diff
@@ -0,0 +1,1564 @@
+            papi_errno = PAPI_EINVAL;
+            goto fn_exit;
+        }
+        insert_event_record(state->gpu_ctl[gpu_id].event_names,
+                            event_names->evts[i].name,
+                            event_names->evts[i].evt_code,
+                            i);
+        LOGDBG("Adding event gpu %d name %s with code %d at pos %d\n",
+            gpu_id, event_names->evts[i].name,
+            event_names->evts[i].evt_code, i);
+    }
+fn_exit:
+    return papi_errno;
+}
+
+static int retrieve_metric_details(NVPA_MetricsContext *pMetricsContext, const char *nv_name,
+                                   char *description, int *numDep, NVPA_RawMetricRequest **pRMR)
+{
+    COMPDBG("Entering.\n");
+    int num_dep, i, len;
+    NVPA_Status nvpa_err;
+
+    if (nv_name == NULL || description == NULL) {
+        return PAPI_EINVAL;
+    }
+
+    NVPW_MetricsContext_GetMetricProperties_Begin_Params getMetricPropertiesBeginParams = {
+        .structSize = NVPW_MetricsContext_GetMetricProperties_Begin_Params_STRUCT_SIZE,
+        .pPriv = NULL,
+        .pMetricsContext = pMetricsContext,
+        .pMetricName = nv_name,
+    };
+    nvpa_err = NVPW_MetricsContext_GetMetricProperties_BeginPtr(&getMetricPropertiesBeginParams);
+
+    if (nvpa_err != NVPA_STATUS_SUCCESS || getMetricPropertiesBeginParams.ppRawMetricDependencies == NULL) {
+        strcpy(description, "Could not get description.");
+        return PAPI_EINVAL;
+    }
+
+    for (num_dep=0; getMetricPropertiesBeginParams.ppRawMetricDependencies[num_dep] != NULL; num_dep++){;}
+
+    NVPA_RawMetricRequest *rmr = (NVPA_RawMetricRequest *) papi_calloc(num_dep, sizeof(NVPA_RawMetricRequest));
+    if (rmr == NULL) {
+        return PAPI_ENOMEM;
+    }
+
+    for (i=0; i<num_dep; i++) {
+        rmr[i].pMetricName = strdup(getMetricPropertiesBeginParams.ppRawMetricDependencies[i]);
+        rmr[i].isolated = 1;
+        rmr[i].keepInstances = 1;
+        rmr[i].structSize = NVPW_MetricsContext_GetMetricProperties_End_Params_STRUCT_SIZE;
+    }
+
+    len = snprintf(description, PAPI_2MAX_STR_LEN, "%s. Units=(%s)",
+                getMetricPropertiesBeginParams.pDescription,
+                getMetricPropertiesBeginParams.pDimUnits);
+    if (len > PAPI_2MAX_STR_LEN) {
+        ERRDBG("String formatting exceeded max string length.\n");
+        return PAPI_ENOMEM;
+    }
+    *numDep = num_dep;
+    *pRMR = rmr;
+    NVPW_MetricsContext_GetMetricProperties_End_Params getMetricPropertiesEndParams = {
+        .structSize = NVPW_MetricsContext_GetMetricProperties_End_Params_STRUCT_SIZE,
+        .pPriv = NULL,
+        .pMetricsContext = pMetricsContext,
+    };
+    NVPW_CALL(NVPW_MetricsContext_GetMetricProperties_EndPtr(&getMetricPropertiesEndParams), return PAPI_EMISC);
+    return PAPI_OK;
+}
+
+static int get_event_names_rmr(NVPA_MetricsContext* pMetricsContext, cuptip_gpu_control_t *gpu_ctl)
+{
+    COMPDBG("Entering.\n");
```

Comment:
This level of logging is not needed but if you really want it in you should have an exit log as well.

Reply:
This is to facilitate easier logging. If an error occurs 3 function calls down from this function, the debug log will tell exactly which path of function calls led to that error. Without it, that would be unclear. The "Exiting" log would serve no such purpose.